File: tekhook-global-sms-quickstart/python/send_otp.py
# send_otp.py — Send a one-time password via Tekhook SMS
# Docs: https://tekhook.co

import random
import os
from dotenv import load_dotenv
from tekhook import TekHook

load_dotenv()

client = TekHook()


def generate_otp(length: int = 6) -> str:
    return "".join([str(random.randint(0, 9)) for _ in range(length)])


def send_otp(phone_number: str) -> dict:
    otp = generate_otp()
    sender = os.environ.get("TEKHOOK_SENDER_ID", "Tekhook")

    print(f"📤 Sending OTP to {phone_number}...")

    result = client.sms.send(
        to=phone_number,
        message=f"Your verification code is: {otp}. Valid for 10 minutes. Do not share this code.",
        sender=sender,
    )

    print(f"✅ OTP sent!")
    print(f"   Message ID : {result['message_id']}")
    print(f"   Status     : {result['status']}")
    print(f"   Delivery   : ~{result.get('delivery_time', 'N/A')}")

    # In production: store otp + expiry in your database
    return {"otp": otp, "message_id": result["message_id"]}


if __name__ == "__main__":
    send_otp("+971585767831")

File: tekhook-global-sms-quickstart/node/send-otp.js
// send-otp.js — Send a one-time password via Tekhook SMS
// Docs: https://tekhook.co

require('dotenv').config();
const { TekHook } = require('./tekhook');

const client = new TekHook(process.env.TEKHOOK_API_KEY);

function generateOTP(length = 6) {
  return Math.floor(Math.pow(10, length - 1) + Math.random() * 9 * Math.pow(10, length - 1)).toString();
}

async function sendOTP(phoneNumber) {
  const otp = generateOTP();

  console.log(`📤 Sending OTP to ${phoneNumber}...`);

  const result = await client.sms.send({
    to: phoneNumber,
    from: process.env.TEKHOOK_SENDER_ID || 'Tekhook',
    message: `Your verification code is: ${otp}. Valid for 10 minutes. Do not share this code.`,
  });

  console.log(`✅ OTP sent!`);
  console.log(`   Message ID : ${result.message_id}`);
  console.log(`   Status     : ${result.status}`);
  console.log(`   Delivery   : ~${result.delivery_time}`);

  // In production: store { otp, expiresAt } in your DB against the user
  return { otp, messageId: result.message_id };
}

// --- Run ---
sendOTP('+971585767831').catch((err) => {
  console.error('❌ Failed to send OTP:', err.message);
  process.exit(1);
});
